packages/amplify_datastore/lib/method_channel_datastore.dart

```diff
@@ -54,19 +54,33 @@ class AmplifyDataStoreMethodChannel extends AmplifyDataStore {
     }
   }
 
-  @override
   Future<void> delete<T extends Model>(T model) async {
-     try {
+    try {
       var modelJson = model.toJson();
       await _channel.invokeMapMethod('delete', <String, dynamic>{
         'modelName': model.instanceType.modelName(),
-        'model': modelJson
+        'serializedModel': modelJson
       });
     } on PlatformException catch (e) {
       throw formatError(e);
     }
   }
 
+  @override
+  Future<void> save<T extends Model>(T model, {QueryPredicate when}) async {
+    try {
+      ModelType<T> modelType = model.instanceType;
+      var methodChannelSaveInput = <String, dynamic>{
+        'modelName': modelType.modelName(),

```

Comment:
nit: To avoid this redirection to get model name, you can simply do `model.runtimeType.toString()`